### PR TITLE
fix(query): fixed issue containers_running_as_root #3412

### LINKED ISSUE #3412: Containers_running_as_root query faild to evaluate
### The query Containers_runs_as_root failed to produce an output
Capture9
Capture10

### for this Yaml file
<pre> 
---
apiVersion: v1
kind: Pod
metadata:
  name: security-context-demo-2
spec:
  securityContext:
    runAsUser: 10
    runAsNonRoot: false
  containers:
  - name: sec-ctx-demo-100
    image: gcr.io/google-samples/node-hello:1.0
    securityContext:
      runAsUser: 0
      runAsNonRoot: false
  - name: sec-ctx-demo-200
    image: gcr.io/google-samples/node-hedwfwllo:1.0
    securityContext:
      runAsUser: 0
      runAsNonRoot: false
</pre>

### Specifications

-   Version: 1.2.3
-   Platform: window

### PR DESCRIPTION
**Pull request by mukeshpilaniya**
Fixes #3412 Containers_running_as_root query failed to evaluate

**Description**

_Added one more corner case_ and fixed already existing issue #3412
<pre>
#if pod runAsNonRoot==true and container runAsNonRoot==true (container not runs as root)
#if pod runAsNonRoot==true and container runAsNonRoot==false 
							#if container runAsUser>0 (container not runs as root)
							#if container runAsUser<=0 (container runs as root) //new added corner case
</pre>

### CODE REVIEW

**Comment by kaplanlior** [comment]
Hi @mukeshpilaniya , first welcome to KICS, we're happy to get a PR from you :+1: 

A bit on the OCD side, but can you fix the typo in the PR/commit? container_ruuiing_as_root -> containers_running_as_root and to close the issue it refers (see [this](https://docs.github.com/en/github/managing-your-work-on-github/managing-your-work-with-issues-and-pull-requests/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) ).

Thanks (:

File: assets/queries/k8s/containers_running_as_root/query.rego [hunk 1]
```diff
@@ -31,98 +35,122 @@ CxPolicy[result] {
...
-		"searchKey": sprintf("metadata.name={{%s}}.%s.%s.{{%s}}.securityContext", [metadata.name, path, types[x], container.name]),
+		"searchKey": sprintf("metadata.name={{%s}}.%s.%s.{{%s}}.securityContext", [metadata.name, path, containerType, container.name]),
 		"issueType": "MissingAttribute",
-		"keyExpectedValue": sprintf("'%s.%s[%d].securityContext.runAsUser' is defined", [path, types[x], j]),
-		"keyActualValue": sprintf("'%s.%s[%d].securityContext.runAsUser' is undefined", [path, types[x], j]),
+		"keyExpectedValue": sprintf("'%s.%s.securityContext.runAsUser' is defined", [path, containerType]),
+		"keyActualValue": sprintf("'%s.%s.securityContext.runAsUser' is undefined", [path, containerType]),
 	}
-}
+}
```
Comment (felipe-avelar) [inline assets/queries/k8s/containers_running_as_root/query.rego:156]: Please add a new line here

**Review by felipe-avelar — CHANGES_REQUESTED** [review]
In general, I think the query changes looks good, I would like just to ask to you to add positive and negative samples covering your new edge case.
If you need some help with this, don't hesitate to ask by @ us at this thread, using gitter or kics@checkmarx.com

**Review by rogeriopeixotocx — APPROVED** [review]
LGTM, thank you @mukeshpilaniya

**Comment by rogeriopeixotocx** [comment]
This PR should fix the integration tests issue https://github.com/Checkmarx/kics/pull/3430

**Comment by rogeriopeixotocx** [comment]
Hi @mukeshpilaniya, now that https://github.com/Checkmarx/kics/pull/3430 is merged can you please update your branch to the master so we can run the PR checks again?

**Comment by mukeshpilaniya** [comment]
> Hi @mukeshpilaniya, now that #3430 is merged can you please update your branch to the master so we can run the PR checks again?

@rogeriopeixotocx, I updated my branch to the master branch. now you can run the Integration test again.

**Comment by joaoReigota1** [comment]
> > Hi @mukeshpilaniya, now that #3430 is merged can you please update your branch to the master so we can run the PR checks again?
> 
> @rogeriopeixotocx, I updated my branch to the master branch. now you can run the Integration test again.

Hi @mukeshpilaniya, I think you forgot to pull the latest changes in the master branch, can you pull and then merge?

**Review by rogeriopeixotocx — APPROVED** [review]
LGTM

**Comment by rogeriopeixotocx** [comment]
@mukeshpilaniya thank you for your contribution

**Pull request merged**